op-batcher/batcher/service_test.go

```diff
@@ -0,0 +1,33 @@
+package batcher
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBatchSubmitter_SignatureGeneration(t *testing.T) {
+	bs, _ := setup(t)
+
+	txdata := emptyTxData
+
+	key, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatalf("failed to generate key pair for batcher: %v", err)
+	}
+	bs.Config.BatcherPrivateKey = key
+	bs.Config.BatcherPublicKey = &key.PublicKey
+
+	// add batcher's signature on txdata sent to L1
+	sig, err := crypto.Sign(crypto.Keccak256(txdata.CallData()), bs.Config.BatcherPrivateKey)
+	require.NoError(t, err)
```

Comment:
See my comment in `driver.go` about potentially replacing this with a helper function to make the implementation and the test consistent.